core/src/main/java/org/infinispan/AdvancedCache.java

```diff
@@ -559,6 +571,27 @@
    CacheSet<CacheEntry<K, V>> cacheEntrySet();
 
    /**
+    * Returns a sequential stream using this Cache as the source. This stream is very similar to using the
+    * {@link CacheStream} returned from the {@link CacheSet#stream()} method of the collection
+    * returned via {@link AdvancedCache#cacheEntrySet()}. The use of this locked stream is that when an entry is
+    * being processed by the user the entry is locked for the invocation preventing a different thread from modifying
+    * it.
+    * <p>
+    * Note that this stream is not supported when using a optimistic transactional or simple cache. Both non transactional and
+    * pessimistc transactions are supported.
+    * <p>
+    * The stream will not share any ongoing transaction the user may have. Code executed by the stream should be treated
+    * as completely independent. That is any operation performed via the stream will require the user to start their
+    * own transaction or will be done intrinsically on the invocation. Note this means that if you lock a key from
+    * the cache it could cause the stream operation to deadlock.
+    * {@link org.infinispan.configuration.cache.ConfigurationBuilder#simpleCache(boolean)} was set to true. This
+    * restriction may be removed in a future version
+    * @return the locked stream
+    * @since 9.1
+    */
+   LockedStream<K, V> lockedStream() throws UnsupportedOperationException;
```

Comment:
I know that I don't, but it makes it more apparent that it can be thrown, which I personally like. Unfortunately in my refactoring of the javadoc I lost the reason as to why this is thrown as you see in the block above re: simple cache. I will fix that.